pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -193,4 +211,29 @@ protected void removeSegmentFromStore(String tableName, String segmentId) {
       LOGGER.info("localDiskDir is not configured, won't delete segment {} from disk", segmentId);
     }
   }
+
+  protected void removeSegmentsFromDeletedDirectory(int deleteRetention) {
```

Comment:
@mayankshriv we can re-factor as needed if we want to re-use this? Would you be ok with that?